src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -155,36 +158,35 @@ def on_revoke(consumer: Consumer, partitions: List[TopicPartition]) -> None:
 
         self.consumer.subscribe([self.topic], on_assign=on_assign, on_revoke=on_revoke)
 
-        try:
-            i = 0
-            while True:
-                message = self.consumer.poll(0.1)
-                if message is None:
-                    continue
+        i = 0
+        while not self.__shutdown_requested:
+            message = self.consumer.poll(0.1)
+            if message is None:
+                continue
 
-                error = message.error()
-                if error is not None:
-                    raise KafkaException(error)
+            error = message.error()
+            if error is not None:
+                raise KafkaException(error)
 
-                i = i + 1
+            i = i + 1
 
-                with sentry_sdk.start_transaction(
-                    op="handle_message",
-                    name="query_subscription_consumer_process_message",
-                    sampled=True,
-                ), metrics.timer("snuba_query_subscriber.handle_message"):
-                    self.handle_message(message)
+            with sentry_sdk.start_transaction(
+                op="handle_message",
+                name="query_subscription_consumer_process_message",
+                sampled=True,
+            ), metrics.timer("snuba_query_subscriber.handle_message"):
+                self.handle_message(message)
 
-                # Track latest completed message here, for use in `shutdown` handler.
-                self.offsets[message.partition()] = message.offset() + 1
+            # Track latest completed message here, for use in `shutdown` handler.
+            self.offsets[message.partition()] = message.offset() + 1
 
-                if i % self.commit_batch_size == 0:
-                    logger.debug("Committing offsets")
-                    self.commit_offsets()
-        except KeyboardInterrupt:
-            pass
+            if i % self.commit_batch_size == 0:
+                logger.debug("Committing offsets")
+                self.commit_offsets()
 
-        self.shutdown()
+        logger.debug("Committing offsets and closing consumer")
+        self.commit_offsets()
```

Comment:
Yeah, doesn't seem like a big deal. Although, since we always complete each iteration of the loop, you could just check `i % self.commit_batch_size != 0`